doc: Generate documentation for the binding examples

rruuaanng opened the pull request:
Generate an rst document with an `Examples` section from the sample nodes written in the examples of the binding.

kartben commented on doc/_scripts/gen_devicetree_rest.py:
```diff
@@ -617,6 +617,15 @@ def print_binding_page(binding, base_names, vnd_lookup, driver_sources,dup_compa
+    ''', string_io)
+    if binding.examples:
+        for example in binding.examples:
+            print(to_code_block(example), file=string_io)
```
add language parameter to to_code_block (defaulting to "none") and pass "dts" so that we have syntax highlighting

kartben commented on doc/_scripts/gen_devicetree_rest.py:
```diff
@@ -819,12 +829,12 @@ def print_block(block, string_io):
     # block, in spaces.
     indent = indent * ' '
-    return ('.. code-block:: none\n\n' +
+    return ('.. code-block:: {language}\n\n' +
```
please test your code :)

```suggestion
    return (f".. code-block:: {language}\n\n" +
```

rruuaanng replied:
```
>>> def to_code(op='none'):
...     return f'Hello {op}'
...
>>> to_code(op='dts')
'Hello dts'
>>> to_code()
'Hello none'
>>> def to_code(op='none'):
...     return f"Hello {op}"
...
>>> to_code(op='dts')
'Hello dts'
>>> to_code()
'Hello none'
>>> to_code(op='dts')
'Hello dts'
```
Honestly, that quote isn't important ;)

rruuaanng committed:
doc: scripts: Add a language parameter to the to_code_block()

Introduce a `language` parameter in `to_code_block()` for syntax
highlighting support (default: None).

Signed-off-by: James Roy <rruuaanng@outlook.com>

pdgendt commented on doc/_scripts/gen_devicetree_rest.py:
```diff
@@ -596,6 +596,16 @@ def print_binding_page(binding, base_names, vnd_lookup, driver_sources,dup_compa
+    print_block('''\
+    Examples
+    ********
+    ''', string_io)
```
Should this be inside the `if binding.examples:` condition? Leaving out the title if there are no examples.

kartben replied:
Ah yeah totally

kartben reviewed (changes requested):
See @pdgendt comment

rruuaanng committed:
doc: Generate documentation for the binding examples

Generate an rst document with an `Examples` section
from the sample nodes written in the examples of the
binding.

Signed-off-by: James Roy <rruuaanng@outlook.com>
